### DISCUSSION TITLE
Ikev2/IPSec Psk support (android )

### DISCUSSION THREAD

**Discussion by scuti-dai** [strongswan/strongswan#1914] [Q&A]
Android: 
I saw that “strongSwan VPN Client for Android” does not support PSK
Link: https://docs.strongswan.org/docs/5.9/os/androidVpnClient.html
So can anyone guide me to do that?

Ios: i customed and it's working now

anyway thank you.

**Comment by salmanmalik-emb** [comment]
We have a requirement to support PSK for android as well. @tobiasbrunner  can you please guide us on this.

**Comment by ciukstar** [comment]
Please add PSK support

**Comment by patrickfrei** [comment]
Does the latest Android version of strongSwan support IKEv2/IPSec PSK? I've an Android 11 mobile phone and unfortunately, the built-in VPN client doesn't offer that option, although my ISP offers only that option.

**Comment by mguinness** [comment]
According to [strongSwan VPN Client for Android](https://docs.strongswan.org/docs/latest/os/androidVpnClient.html) this is intentional.

> PSK authentication is not supported, as it is potentially very dangerous because the client might send the hash of a weak password to a rogue VPN server. Thus we prefer EAP authentication where the server is first authenticated by an X.509 certificate and only afterwards the client uses its password.

**Comment by NishchayRajput** [comment]
Can we implement the the functionality? Please provide the relatable documentation for the implementation.

**Comment by celesteking** [comment]
Android 11's built-in VPN client supports PSK. 

> Can we implement the the functionality? 

How many thousand $$ are you willing to pay?

**Reply by NishchayRajput** [reply]
> How many thousand $$ are you willing to pay?

we don't want to use the inbuilt Vpn. We are designing something different in which we want the psk functionality in strongswan android app
